File: client/src/enums/appearance/ageRange.ts
/**
 * Represents age range visualization options for image generation.
 * Used in the User model to store appearance preferences.
 */
export enum AgeRange {
  TWENTIES = "twenties",
  THIRTIES = "thirties",
  FORTIES = "forties",
  FIFTIES = "fifties",
  SIXTY_PLUS = "sixty_plus",
}

/**
 * Maps age range values to their human-readable display names
 */
export const AGE_RANGE_DISPLAY_NAMES: Record<AgeRange, string> = {
  [AgeRange.TWENTIES]: "Young Adult (20s)",
  [AgeRange.THIRTIES]: "In Their 30s",
  [AgeRange.FORTIES]: "In Their 40s",
  [AgeRange.FIFTIES]: "Middle-Aged (50s)",
  [AgeRange.SIXTY_PLUS]: "Mature (60+)",
};

File: server/enums/appearance/age_range.py
from django.db import models


class AgeRange(models.TextChoices):
    """
    Enum for age range visualization options in image generation.
    """

    TWENTIES = "twenties", "Young Adult (20s)"
    THIRTIES = "thirties", "In Their 30s"
    FORTIES = "forties", "In Their 40s"
    FIFTIES = "fifties", "Middle-Aged (50s)"
    SIXTY_PLUS = "sixty_plus", "Mature (60+)"

    def __str__(self) -> str:
        """
        Get a human-readable string representation of the age range.
        """
        return self.label
